src/Kernel.sol

```diff
@@ -85,7 +89,9 @@ contract Kernel is IAccount, EIP712, Compatibility, KernelStorage {
         external
         returns (uint256 validationData)
     {
-        require(msg.sender == address(entryPoint), "account: not from entryPoint");
+        if (msg.sender != address(entryPoint)) {
```

Comment:
If we do custom error on validation, Entrypoint does not return error string which increases difficulty to debug the validation phase, i do agree that this will reduce gas fee but i think debugging is another feature that we should not compromise. But i want to hear your thought regarding this